### PR TITLE
treewide: migrate packages to by-name hierarchy

### PR DESCRIPTION
**Pull request by AndersonTorres**
## Description of changes

The ones maintained by @AndersonTorres

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mohe2015** [comment]
I think it was discussed at NixCon that we should move to the new schema using automation (where possible) to avoid the overhead of many PRs. You also did some cleanup so in this case this is probably fine.

**Comment by AndersonTorres** [comment]
@ofborg build pkgsLLVM.mksh mksh

**Comment by infinisil** [comment]
Yeah so I will work on updating https://github.com/NixOS/nixpkgs/pull/211832 next, which is the automated mass migration. I wouldn't want people to have to do this manually to save on both reviewer and contributor time.

I think it's fine if you happen to clean up the code or update the versions anyways though. Since you did the work here already I guess it's fine, but others should definitely not be encouraged to do this manually until the automated migration is done (it doesn't work automatically for all packages) :slightly_smiling_face:

**Comment by AndersonTorres** [comment]
OK, I will stop this thread for now and concentrate in the version bumps.

**Comment by AndersonTorres** [comment]
@ofborg build xorriso dosbox-staging

File: pkgs/by-name/sc/scimark/package.nix [hunk 1]
```diff
@@ -20,17 +20,21 @@ stdenv.mkDerivation rec {
   dontConfigure = true;
 
   installPhase = ''
-    install -d $out/bin/
-    install scimark4 $out/bin/
+    runHook preInstall
+
+    install -Dm755 scimark4 -g $out/bin/
```
Comment (Artturin) [inline pkgs/by-name/sc/scimark/package.nix:25]: `install: missing destination file operand after 'scimark4'`

**Comment by infinisil** [comment]
Are you intending on having this reviewed and merged after all? Or are you using this as a testing ground before splitting it into smaller cleanup PRs? If so this should probably be turned into a draft.

**Pull request merged**